### PR TITLE
Feat(eos_cli_config_gen): Support of next_hop_unchanged under EVPN address family

### LINKED ISSUE #3102: Feat(eos_cli_config_gen): Support of next_hop_unchanged under EVPN address family
### Enhancement summary

Looking at the current configuration data model, next_hop_unchanged is only available under router_bgp/peer_groups hierarchy while it is also possible to be configured under router_bgp/address_family_evpn for example (as configured for my customer's IP Fabric spines)

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

<pre>
router bgp xxxx
...
   address-family evpn
      bgp next-hop-unchanged
      neighbor LEAVES-EVPN activate
</pre>

### Describe the solution you would like

- Change schema files
- Change router-bgp.j2 template

### Describe alternatives you have considered

The alternative of changing the customer's configuration to apply the next_hop_unchanged at peer_group level is exposing a limitation which shouldn't be difficult to implement.

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by mmaaloul**
## Change Summary

Permit the explicit configuration of next-hop-unchanged knob under address-family evpn

## Related Issue(s)

Fixes #3102

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
1/ Modified schema fragment "router_bgp" and template "router-bgp" accordingly
2/ Modified device documentation template accordingly
3/ Added a molecule test (new host "router-bgp-evpn-route-server") as the knob is specific to route server role 

## How to test
The knob is present under address-family evpn in the generated intended configuration 

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/hosts.ini [hunk 1]
```diff
@@ -107,6 +107,7 @@ route-maps
 router-bfd
 router-bgp-base
 router-bgp-evpn
+router-bgp-evpn-route-server
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/hosts.ini]: could we do without a new device in molecule to just test "one knob" ? (I know adding it to the existing `router-bgp-evpn` might feel wrong but we really just need to check we generate the knob)

**Review by gmuloc — COMMENTED** [review]
Overall LGTM - just one comment regarding adding a new host to molecule only for one line. @ClausHolbechArista is it ok here to put it in `router-bgp-evpn` even if the config "might be wrong"

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/hosts.ini (hunk 1 above)
Comment (mmaaloul) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/hosts.ini]: Done although it is incompatible with the existing evpn config of "host" router-bgp-evpn

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2 [hunk 2]
```diff
@@ -577,6 +577,9 @@ router bgp {{ router_bgp.as }}
 {%         if router_bgp.address_family_evpn.route.import_match_failure_action is arista.avd.defined('discard') %}
       route import match-failure action discard
 {%         endif %}
+{%         if router_bgp.address_family_evpn.next_hop_unchanged is arista.avd.defined(true) %}
+      bgp next-hop-unchanged
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2]: Please confirm that the configuration order on EOS has this as the last command compared to the others under evpn.

**Review by ClausHolbechArista — APPROVED** [review]
LGTM. Note my comment on checking the order on EOS. We try to follow the ordering on EOS for anything new.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2 (hunk 2 above)
Comment (mmaaloul) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2]: According to the output of EOS command "show cli commands all-modes verbatim | grep RouterBgpBaseAfEvpn", the "next-hop-unchanged" should be preceding all the remaining knobs of address-family evpn. Hence reordering at the J2 template level

**Review by gmuloc — APPROVED** [review]

**Pull request merged**